### ISSUE TITLE
Clarity on TC handling of Additional Child Tax Credit

### ISSUE THREAD

**Issue by evtedeschi3** [open-source-economics/Tax-Calculator#1665]
My impression reading the tc documentation is that the model by default treats only those families with 3 or more children as eligible for the Additional Child Tax Credit.

If that impression is wrong, then this thread should hard stop here.

But if that's the way tc treats it, that's not what the law says. See e.g. [here](https://www.hrblock.com/tax-center/filing/credits/child-tax-credit/):

> **Additional child tax credit**
>
>If your child tax credit is limited, you might be able to claim the additional child tax credit. To qualify, **one** of these must apply _[emphasis mine]_:
>
>- Your earned income must be more than $3,000 for 2017.
>- You must have three or more qualifying children.
>
>If you have at least one qualifying child, you can claim a credit of up to 15% of the amount your earned income is more than $3,000. If you have three or more qualifying children, you can either:
>
> - Claim a refundable credit if the net Social Security and Medicare tax you paid is more than your Earned Income Credit (EIC), if any
> - Use the 15% method described earlier

Also, if you walk through the [IRS worksheet for the CTC and ACTC](https://www.irs.gov/pub/irs-pdf/f8812.pdf), it's clear that having 3 or more kids does not disqualify you, it just bumps you into a different eligibility worksheet.

So having 1 or 2 kids simply means your ACTC must be calculated using the 15%*(earned income - $3,000) approach, not the approach that allows a credit for FICA taxes in excess of the EITC.

Wanted to clarify because recent tax reform proposals have tinkered with the ACTC, which can have a meaningful impact particularly on low-income parents.

**Comment by martinholmer** [comment]
@evtedeschi3,  Thanks for your continued interest in Tax-Calculator.

Your question in issue #1665 about the Additional Child Tax Credit (ACTC) is a good one because it reveals either poor documentation or incorrect tax-calculating logic or both.

Let's figure out, in this issue, what needs to be fixed.  

**First**, I'll show you below the ACTC calculation logic (which you will see is dated, and possibly out-of-date if there have been changes in the ACTC over the past few years).  The Python code, which is strongly oriented to using variable names that IRS-SOI staff, but few others, know by heart, is a little hard to follow but the outlines of the logic should be apparent.  I particular, my quick glance at the code, suggests that filing units with fewer than three children are eligible for the ACTC.  But I would appreciate your review of the logic.

**Second**, I hope you can point us to the place(s) in the user documentation that led you to raise this issue.

Here is the Python function that computes the ACTC amount:
```
def AdditionalCTC(n24, prectc, earned, c07220, ptax_was,
                  ACTC_Income_thd, ACTC_rt, ACTC_ChildNum,
                  c03260, e09800, c59660, e11200, c11070, nu05,
                  ACTC_rt_bonus_under5family):
    """
    Calculates Additional (refundable) Child Tax Credit, c11070
    """
    c82925 = 0.
    c82930 = 0.
    c82935 = 0.
    c82880 = 0.
    c82885 = 0.
    c82890 = 0.
    c82900 = 0.
    c82905 = 0.
    c82910 = 0.
    c82915 = 0.
    c82920 = 0.
    c82937 = 0.
    c82940 = 0.
    c11070 = 0.
    # Part I of 2005 Form 8812
    if n24 > 0:
        c82925 = prectc
        c82930 = c07220
        c82935 = c82925 - c82930
        # CTC not applied to tax
        c82880 = max(0., earned)
        c82885 = max(0., c82880 - ACTC_Income_thd)
        # Accomodate ACTC rate bonus for families with children under 5
        if nu05 == 0:
            ACTC_rate = ACTC_rt
        else:
            ACTC_rate = ACTC_rt + ACTC_rt_bonus_under5family
        c82890 = ACTC_rate * c82885
    # Part II of 2005 Form 8812
    if n24 >= ACTC_ChildNum and c82890 < c82935:
        c82900 = 0.5 * ptax_was
        c82905 = c03260 + e09800
        c82910 = c82900 + c82905
        c82915 = c59660 + e11200
        c82920 = max(0., c82910 - c82915)
        c82937 = max(c82890, c82920)
    # Part II of 2005 Form 8812
    if n24 > 0 and n24 <= 2 and c82890 > 0:
        c82940 = min(c82890, c82935)
    if n24 > 2:
        if c82890 >= c82935:
            c82940 = c82935
        else:
            c82940 = min(c82935, c82937)
    c11070 = c82940
    return c11070
```

**Comment by evtedeschi3** [comment]
Thanks @martinholmer. 

> I hope you can point us to the place(s) in the user documentation that led you to raise this issue.

Mainly I was looking at the `current_law_policy.json` [description](https://github.com/open-source-economics/Tax-Calculator/blob/master/taxcalc/current_law_policy.json) for `_ACTC_ChildNum`:

> Families with this number of qualified children or more may qualify for the Additional Child Tax Credit, which is a partially refundable credit that supplements the Child Tax Credit for some taxpayers

I read this as "no one below this parameter qualifies" rather than "families below this parameter are constrained to only one possible credit formula"

In terms of the Python code, yes, I agree with you that families with fewer than `ACTC_ChildNum` kids appear to still qualify. 

On that note: why is the "2" here hardcoded? Should that instead be `ACTC_ChildNum`?

```
# Part II of 2005 Form 8812
if n24 > 0 and n24 <= 2 and c82890 > 0:
```

**Comment by martinholmer** [comment]
@evtedeschi3,  Thanks for the code review and thanks especially for identifying needed improvements to both the documentation and to the tax-calculating code.  
I'll open a pull request to improve things along the lines you suggest.  
It would be appreciated if you could review the proposed changes.   
Thanks again.

**Comment by martinholmer** [comment]
Merge of pull request #1666 has resolved issue #1665, which is now being closed.
